<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Source: api.js</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Source: api.js</h1>

    



    
    <section>
        <article>
            <pre class="prettyprint source linenums"><code>import axios from 'axios';

/**
 * Funkcja do uwierzytelniania dwuskładnikowego
 * @param {FormData} formData2FA - Dane uwierzytelniania dwuskładnikowego
 * @returns {Promise&lt;object>} - Obiekt odpowiedzi po uwierzytelnieniu
 * @throws {Error} - Błąd podczas uwierzytelniania dwuskładnikowego
 */
export const authenticate2FA = async (formData2FA) => {
  try{
    const addr = `${process.env.EXPO_PUBLIC_API_URL}/main/authenticate/`
    console.info(addr)
    const response = await axios.post(addr,formData2FA);
    return response.data;
  }catch(error)
  {
    throw new Error(error.response.data.error);
  }
}

/**
 * Funkcja do rejestracji użytkownika
 * @param {FormData} formDataRegister - Dane rejestracyjne użytkownika
 * @returns {Promise&lt;object>} - Obiekt odpowiedzi po rejestracji
 * @throws {Error} - Błąd podczas rejestracji użytkownika
 */
export const registerUser = async (formDataRegister) => {
  try {
    const addr = `${process.env.EXPO_PUBLIC_API_URL}/main/register/`
    console.info(addr)
    const response = await axios.post(addr,formDataRegister);
    return response.data;
  } catch (error) {
    throw new Error(error.response.data.error);
  }
};

/**
 * Funkcja do logowania użytkownika
 * @param {FormData} formDataLogin - Dane logowania użytkownika
 * @returns {Promise&lt;object>} - Obiekt odpowiedzi po zalogowaniu
 * @throws {Error} - Błąd podczas logowania użytkownika
 */
export const loginUser = async (formDataLogin) => {
  try {
    const addr = `${process.env.EXPO_PUBLIC_API_URL}/main/login/`
    console.info(addr)
    const response = await axios.post(addr,formDataLogin);
    return response.data;
  } catch (error) {
    throw new Error(error.response.data.error);
  }
};

/**
 * Obiekt do wykonywania żądań HTTP do API
 */
const api = axios.create({
  baseURL: process.env.EXPO_PUBLIC_API_URL,
  withCredentials: true
})
export default api;</code></pre>
        </article>
    </section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Modules</h3><ul><li><a href="module-App.html">App</a></li></ul><h3>Global</h3><ul><li><a href="global.html#api">api</a></li><li><a href="global.html#authenticate2FA">authenticate2FA</a></li><li><a href="global.html#loginUser">loginUser</a></li><li><a href="global.html#registerUser">registerUser</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 4.0.2</a> on Thu Nov 30 2023 12:08:00 GMT+0100 (czas środkowoeuropejski standardowy)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>
